### ISSUE TITLE
Style `console` outputs using ANSI Escape Codes

### ISSUE THREAD

**Issue by signorettif** [kettanaito/msw#1795]
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

Hey team, 

thank you for the great work on this library! 

I think we could switch some of console styling ([example](https://github.com/mswjs/msw/blob/a54138a62a7e89bd5f768d369e6b5bc4c46686ae/src/core/handlers/GraphQLHandler.ts#L182-L189)) to use [ANSI Escape Codes](https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797) to broaden the compatibility of the output. While the current approach works well when previewed in Chromium-like Dev Tools—piping the console output using tools such as `puppeteer` might not preserve it.

A similar argument can be proposed to also pre-format any string in console outputs instead of relying on format specifiers.

Example when parsing the console output using `puppeteer.ConsoleMessage.text` function below:

```
[MSW] %s %s %s (%c%s%c) 19:50:06 POST https://app.myDomain.com/some-route color:#69AB32 200 OK color:inherit
```

Happy to make a contribution if there's appetite for this change.

**Comment by kettanaito** [comment]
Hi, @signorettif. Thanks for proposing this. 

Are you aware if this isn't simply a bug in Puppeteer? The way it works is it sends the messages between the browser and the test (Node.js) processes, so the console formatting gets lost (in fact, it gets interpreted _literally_, preserving the format placeholders instead of resolving them). 

Do you perchance trying to capture a console message in the browser, send it to your test, and _assert_ it?

**Comment by kettanaito** [comment]
I've looked into it and I think this is actually a bug. On this line:

https://github.com/mswjs/msw/blob/9e9c5a14f068837d7529793abc07181dd2fe1b3b/src/core/handlers/HttpHandler.ts#L155-L163

The entire console string should be wrapped in `devUtils.formatMessage()`. That function does precisely what you propose: renders a template string with positionals into a single string.

**Comment by kettanaito** [comment]
## Released: v2.0.2 🎉

This has been released in v2.0.2!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.0.2)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.0.2)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
